src/page/mod.rs

```diff
@@ -63,6 +63,27 @@ pub enum CacheEntry<M> {
 
 unsafe impl<M: Send> Send for CacheEntry<M> {}
 
+#[derive(Debug, Clone)]
+pub struct CasKey<P> {
+    ptr: *const stack::Node<CacheEntry<P>>,
+    tag: usize,
+}
+
+impl<'s, P> From<Ptr<'s, stack::Node<CacheEntry<P>>>> for CasKey<P> {
+    fn from(ptr: Ptr<'s, stack::Node<CacheEntry<P>>>) -> CasKey<P> {
+        CasKey {
+            ptr: ptr.as_raw(),
+            tag: ptr.tag(),
+        }
+    }
+}
```

Comment:
this lets coco shared atomic `Ptr`'s be converted to a `CasKey` as it crosses the boundary from coco-scoped land to the external users of the pagecache API (which should not be able to dereference any shared pointers)